README.md
# Geoao

A package that uses Geoao API developed by ATECH, aiming to expose all geolocation data referring to Angola.

## MainClass ✨ 
```dart
/// GEOAO
///
/// Geolocation Angola - Geoao
/// A package that uses geoao API aiming to expose all geolocation data referring to Angola
class Geoao {
  ProvinceService _provinceService;
  CountyService _countyService;

  Geoao() {
    _provinceService = ProvinceService();
    _countyService = CountyService();
  }

  /// getProvinces
  ///
  /// Get all provinces of Angola
  Future<ProvincesModel> getProvinces() async =>
      await _provinceService.getProvinces();

  /// getCounties
  ///
  /// Get all counties from a province [provinceId] of Angola
  Future<List<CountyModel>> getCounties(int provinceId) async =>
      await _countyService.getCounties(provinceId);

  /// getCountyById
  ///
  /// Get specific county [countyId] of Angola
  Future<CountyModel> getCountyById(int countyId) async =>
      await _countyService.getCountyById(countyId);
}
```

## Usage 🤳
```dart
import 'package:geoao/geoao.dart';

_exampleUsage() async {
  Geoao _geoao;
  
  ProvincesModel provincesModel;
  List<CountyModel> counties;
  CountyModel countyModel;

  int idProvince = 1;
  int idCounty = 1;

  provincesModel = await _geoao.getProvinces();
  counties = await _geoao.getCounties(idProvince);
  countyModel = await _geoao.getCountyById(idCounty);
}
```

### MIT License


CHANGELOG.md
## [0.0.1+1] - 2021-03-16

* Solved errors when loading local files 
* Adjusted documentation according dartdocs

## [0.0.1] - 2021-03-16

* Initial version
* Geoao has getProvinces method
* Geoao has getCounties method
* Geoao has getCountyByProvince method
* Geoao loads only from local asset



test/geoao_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:geoao/geoao.dart';

void main() {
  test('test provinces and counties assets', () async {
    TestWidgetsFlutterBinding.ensureInitialized();

    final geoao = Geoao();
    expect((await geoao.getCounties(1))[0].countyName, 'Belas');
    // expect(calculator.addOne(2), 3);
    // expect(calculator.addOne(-7), -6);
    // expect(calculator.addOne(0), 1);
  });
}


lib/geoao.dart
library geoao;

export 'package:geoao/src/data/models/county_model.dart';
export 'package:geoao/src/data/models/province_model.dart';

import 'package:geoao/src/data/models/county_model.dart';
import 'package:geoao/src/data/models/province_model.dart';
import 'package:geoao/src/data/services/county_service.dart';
import 'package:geoao/src/data/services/province_service.dart';

/// GEOAO
///
/// Geolocation Angola - Geoao
/// A package that uses geoao API aiming to expose all geolocation data referring to Angola
class Geoao {
  /// The province service provider
  ProvinceService _provinceService;

  /// The county service provider
  CountyService _countyService;

  Geoao() {
    _provinceService = ProvinceService();
    _countyService = CountyService();
  }

  /// getProvinces
  ///
  /// Get all provinces of Angola
  Future<ProvincesModel> getProvinces() async =>
      await _provinceService.getProvinces();

  /// getCounties
  ///
  /// Get all counties from a province [provinceId] of Angola
  Future<List<CountyModel>> getCounties(int provinceId) async =>
      await _countyService.getCounties(provinceId);

  /// getCountyById
  ///
  /// Get specific county [countyId] of Angola
  Future<CountyModel> getCountyById(int countyId) async =>
      await _countyService.getCountyById(countyId);
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:geoao/geoao.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  Geoao _geoao;
  ProvincesModel provincesModel;
  List<CountyModel> counties;
  CountyModel countyModel;

  @override
  void initState() {
    super.initState();
    _getProvinces();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        children: [],
      ),
    );
  }

  void _getProvinces() async {
    provincesModel = await _geoao.getProvinces();

    setState(() {});
  }

  void _getCounties(int idProvince) async {
    counties = await _geoao.getCounties(idProvince);

    setState(() {});
  }

  void _getCountyById(int idCounty) async {
    countyModel = await _geoao.getCountyById(idCounty);

    setState(() {});
  }
}


lib/src/data/json/provinces.dart
/// Provinces Angola data according to MAT - Ministério da Administração e Território [2019]
var provinces = [
  {"mat_id": "AOLDA05", "id_province": 1,"province_name":"Luanda","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOBGO04", "id_province": 2,"province_name":"Bengo","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOBIE12", "id_province": 3,"province_name":"Bié","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOCDA01", "id_province": 4,"province_name":"Cabinda","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOBLA14", "id_province": 5,"province_name":"Benguela","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOCCU18", "id_province": 6,"province_name":"Cuando Cubango","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOCNE17", "id_province": 7,"province_name":"Cunene","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOCSU07", "id_province": 8,"province_name":"Cuanza Sul","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOCNO06", "id_province": 9,"province_name":"Cuanza-Norte","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOHBO13", "id_province": 10,"province_name":"Huambo","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOHLA16", "id_province": 11,"province_name":"Huíla","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOLNO09", "id_province": 12,"province_name":"Lunda-Norte","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOLSU10", "id_province": 13,"province_name":"Lunda-Sul","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOMJE08", "id_province": 14,"province_name":"Malanje","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOMCO11", "id_province": 15,"province_name":"Moxico","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AONBE15", "id_province": 16,"province_name":"Namibe","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOUGE03", "id_province": 17,"province_name":"Uíge","created_at":"2019-08-29 11:41:08"},
  {"mat_id": "AOZRE02", "id_province": 18,"province_name":"Zaire","created_at":"2019-08-29 11:41:08"}
];


lib/src/data/models/county_model.dart
class CountiesModel {
  CountiesModel({
    this.counties,
  });

  List<CountyModel> counties;

  factory CountiesModel.fromJson(List<dynamic> json) => CountiesModel(
    counties: json == null
        ? null
        : List<CountyModel>.from(
        json.map((x) => CountyModel.fromJson(x))),
  );

  List<dynamic> toJson() => [
    counties == null
        ? null
        : List<dynamic>.from(counties.map((x) => x.toJson())),
  ];
}

class CountyModel {
  CountyModel({
    this.id,
    this.idCounty,
    this.idProvince,
    this.countyName,
    this.createdAt,
  });

  String id;
  int idCounty;
  int idProvince;
  String countyName;
  DateTime createdAt;

  factory CountyModel.fromJson(Map<String, dynamic> json) => CountyModel(
    id: json["_id"] == null ? null : json["_id"],
    idCounty: json["id_county"] == null ? null : json["id_county"],
    idProvince: json["id_province"] == null ? null : json["id_province"],
    countyName: json["county_name"] == null ? null : json["county_name"],
    createdAt: json["created_at"] == null
        ? null
        : DateTime.parse(json["created_at"]),
  );

  Map<String, dynamic> toJson() => {
    "_id": id == null ? null : id,
    "id_county": idCounty == null ? null : idCounty,
    "id_province": idProvince == null ? null : idProvince,
    "county_name": countyName == null ? null : countyName,
    "created_at": createdAt == null ? null : createdAt.toIso8601String(),
  };
}


lib/src/data/models/province_model.dart
class ProvincesModel {
  ProvincesModel({
    this.provinces,
  });

  List<ProvinceModel> provinces;

  factory ProvincesModel.fromJson(List<dynamic> json) =>
      ProvincesModel(
        provinces: json == null
            ? null
            : List<ProvinceModel>.from(
            json.map((x) => ProvinceModel.fromJson(x))),
      );

  List<dynamic> toJson() => [
    provinces == null
        ? null
        : List<dynamic>.from(provinces.map((x) => x.toJson())),
  ];
}

class ProvinceModel {
  ProvinceModel({
    this.id,
    this.matId,
    this.idProvince,
    this.provinceName,
    this.createdAt,
  });

  String id;
  String matId;
  int idProvince;
  String provinceName;
  DateTime createdAt;

  factory ProvinceModel.fromJson(Map<String, dynamic> json) => ProvinceModel(
    id: json["_id"] == null ? null : json["_id"],
    matId: json["mat_id"] == null ? null : json["mat_id"],
    idProvince: json["id_province"] == null ? null : json["id_province"],
    provinceName:
    json["province_name"] == null ? null : json["province_name"],
    createdAt: json["created_at"] == null
        ? null
        : DateTime.parse(json["created_at"]),
  );

  Map<String, dynamic> toJson() => {
    "_id": id == null ? null : id,
    "mat_id": matId == null ? null : matId,
    "id_province": idProvince == null ? null : idProvince,
    "province_name": provinceName == null ? null : provinceName,
    "created_at": createdAt == null ? null : createdAt.toIso8601String(),
  };
}


lib/src/data/services/county_service.dart
import 'dart:convert';

import 'package:flutter/foundation.dart';
import 'package:geoao/src/data/json/counties.dart';
import 'package:geoao/src/data/models/county_model.dart';

/// A service tha provides data for counties
class CountyService {

  /// Loads counties from local file [counties.json]
  ///
  /// Throws [Exception] if there was any error loading the file
  Future<String> _loadCountiesFromAsset() async {
    try {
      return jsonEncode(counties);
    } catch(e, stackTrace) {
      print("ERROR loading from assets");
      print(e);
      print(stackTrace);

      throw Exception("Error loading assets from device. Please check available space.");
    }
  }

  /// Gets the counties by province id
  ///
  /// Throws [FlutterError] if there was any error loading the counties.
  Future<CountiesModel> _loadCounties() async {
    var res = await _loadCountiesFromAsset();

    if(res != null) {
      return CountiesModel.fromJson(json.decode(res));
    }

    return null;
  }

  /// Gets the counties by province id
  ///
  /// Throws [FlutterError] if there was any error loading the counties.
  Future<List<CountyModel>> getCounties(int provinceId) async {
    var countiesModel = await _loadCounties();

    if(countiesModel != null) {

      return countiesModel.counties.where((e) => e.idProvince == provinceId).toList();

    } else {
      throw FlutterError("Error loading counties.");
    }
  }

  /// Get county by its id
  ///
  /// Throws [FlutterError] if there was any error loading the county data.
  Future<CountyModel> getCountyById(int countyId) async {
    var countiesModel = await _loadCounties();

    if(countiesModel != null) {

      return countiesModel.counties.singleWhere((e) => e.idCounty == countyId, orElse: () => null);

    } else {
      throw FlutterError("Error loading county.");
    }
  }

  /// TODO: Load counties from API
  ///
  /// Throws [Exception] if there was any error loading the file.
  /// Using the [http] package
}


lib/src/data/services/province_service.dart
import 'dart:convert';

import 'package:flutter/cupertino.dart';
import 'package:geoao/src/data/json/provinces.dart';
import 'package:geoao/src/data/models/province_model.dart';

/// A service tha provides data for provinces
class ProvinceService {

  /// Loads provinces from local file [provinces.json]
  ///
  /// Throws [Exception] if there was any error loading the file
  Future<String> _loadProvincesFromAsset() async {
    try {
      return jsonEncode(provinces);
    } catch(e, stackTrace) {
      print("ERROR loading from assets");
      print(e);
      print(stackTrace);

      throw Exception("Error loading assets from device. Please check available space");
    }

  }

  /// Loads provinces
  ///
  /// Throws [FlutterError] if there was any error loading the provinces.
  Future<ProvincesModel> _loadProvinces() async {
    var res = await _loadProvincesFromAsset();

    if(res != null) {
      return ProvincesModel.fromJson(json.decode(res));
    }

    return null;
  }

  /// Gets the provinces
  ///
  /// Throws [FlutterError] if there was any error loading the provinces.
  getProvinces() async {
    var provincesModel = await _loadProvinces();

    if(provincesModel != null) {

      return provincesModel;

    } else {
      throw FlutterError("Error loading provinces.");
    }
  }
}


lib/src/data/json/counties.dart
/// Counties Angola data according to MAT - Ministério da Administração e Território [2019]
var counties = [
  {"id_county": 1,"id_province": 2,"county_name":"Caxito"},
  {"id_county": 2,"id_province": 3,"county_name":"Ambriz"},
  {"id_county": 3,"id_province": 2,"county_name":"Bula Atumba"},
  {"id_county": 4,"id_province": 2,"county_name":"Dande"},
  {"id_county": 5,"id_province": 2,"county_name":"Dembos"},
  {"id_county": 6,"id_province": 2,"county_name":"Nambuangongo"},
  {"id_county": 7,"id_province": 2,"county_name":"Pango Aluquém"},
  {"id_county": 8,"id_province": 5,"county_name":"Benguela"},
  {"id_county": 9,"id_province": 5,"county_name":"Balombo"},
  {"id_county": 10,"id_province": 5,"county_name":"Baía Farta"},
  {"id_county": 11,"id_province": 5,"county_name":"Benguela"},
  {"id_county": 12,"id_province": 5,"county_name":"Bocoio"},
  {"id_county": 13,"id_province": 5,"county_name":"Caimbambo"},
  {"id_county": 14,"id_province": 5,"county_name":"Catumbela"},
  {"id_county": 15,"id_province": 5,"county_name":"Chongorói"},
  {"id_county": 16,"id_province": 5,"county_name":"Cubal"},
  {"id_county": 17,"id_province": 5,"county_name":"Ganda"},
  {"id_county": 18,"id_province": 5,"county_name":"Lobito"},
  {"id_county": 19,"id_province": 3,"county_name":"Andulo"},
  {"id_county": 20,"id_province": 3,"county_name":"Camacupa"},
  {"id_county": 21,"id_province": 3,"county_name":"Catabola"},
  {"id_county": 22,"id_province": 3,"county_name":"Chinguar"},
  {"id_county": 23,"id_province": 3,"county_name":"Chitembo"},
  {"id_county": 24,"id_province": 3,"county_name":"Cuemba"},
  {"id_county": 25,"id_province": 3,"county_name":"Cunhinga"},
  {"id_county": 26,"id_province": 3,"county_name":"Kuito"},
  {"id_county": 27,"id_province": 3,"county_name":"Nharea"},
  {"id_county": 28,"id_province": 4,"county_name":"Belize"},
  {"id_county": 29,"id_province": 4,"county_name":"Buco-Zau"},
  {"id_county": 30,"id_province": 4,"county_name":"Cabinda"},
  {"id_county": 31,"id_province": 4,"county_name":"Cacongo"},
  {"id_county": 32,"id_province": 6,"county_name":"Calai"},
  {"id_county": 33,"id_province": 6,"county_name":"Cuangar"},
  {"id_county": 34,"id_province": 6,"county_name":"Cuchi"},
  {"id_county": 35,"id_province": 6,"county_name":"Cuito Cuanavale"},
  {"id_county": 36,"id_province": 6,"county_name":"Dirico"},
  {"id_county": 37,"id_province": 6,"county_name":"Longa"},
  {"id_county": 38,"id_province": 6,"county_name":"Mavinga"},
  {"id_county": 39,"id_province": 6,"county_name":"Menongue"},
  {"id_county": 40,"id_province": 6,"county_name":"Nancova"},
  {"id_county": 41,"id_province": 6,"county_name":"Rivungo"},
  {"id_county": 42,"id_province": 7,"county_name":"Cahama"},
  {"id_county": 43,"id_province": 7,"county_name":"Cuanhama"},
  {"id_county": 44,"id_province": 7,"county_name":"Curoca"},
  {"id_county": 45,"id_province": 7,"county_name":"Cuvelai"},
  {"id_county": 46,"id_province": 7,"county_name":"Namacunde"},
  {"id_county": 47,"id_province": 7,"county_name":"Ombadja"},
  {"id_county": 48,"id_province": 10,"county_name":"Bailundo"},
  {"id_county": 49,"id_province": 10,"county_name":"Catchiungo"},
  {"id_county": 50,"id_province": 10,"county_name":"Caála"},
  {"id_county": 51,"id_province": 10,"county_name":"Ekunha"},
  {"id_county": 52,"id_province": 10,"county_name":"Huambo"},
  {"id_county": 53,"id_province": 10,"county_name":"Londuimbale"},
  {"id_county": 54,"id_province": 10,"county_name":"Longonjo"},
  {"id_county": 55,"id_province": 10,"county_name":"Mungo"},
  {"id_county": 56,"id_province": 10,"county_name":"Tchicala-Tcholoanga"},
  {"id_county": 57,"id_province": 10,"county_name":"Tchindjenje"},
  {"id_county": 58,"id_province": 10,"county_name":"Ucuma"},
  {"id_county": 59,"id_province": 11,"county_name":"Caconda"},
  {"id_county": 60,"id_province": 11,"county_name":"Cacula"},
  {"id_county": 61,"id_province": 11,"county_name":"Caluquembe"},
  {"id_county": 62,"id_province": 11,"county_name":"Chiange"},
  {"id_county": 63,"id_province": 11,"county_name":"Chibia"},
  {"id_county": 64,"id_province": 11,"county_name":"Chicomba"},
  {"id_county": 65,"id_province": 11,"county_name":"Chipindo"},
  {"id_county": 66,"id_province": 11,"county_name":"Cuvango"},
  {"id_county": 67,"id_province": 11,"county_name":"Humpata"},
  {"id_county": 68,"id_province": 11,"county_name":"Jamba"},
  {"id_county": 69,"id_province": 11,"county_name":"Lubango"},
  {"id_county": 70,"id_province": 11,"county_name":"Matala"},
  {"id_county": 71,"id_province": 11,"county_name":"Quilengues"},
  {"id_county": 72,"id_province": 11,"county_name":"Quipungo"},
  {"id_county": 74,"id_province": 9,"county_name":"Ambaca"},
  {"id_county": 75,"id_province": 9,"county_name":"Banga"},
  {"id_county": 76,"id_province": 9,"county_name":"Bolongongo"},
  {"id_county": 77,"id_province": 9,"county_name":"Cambambe"},
  {"id_county": 78,"id_province": 9,"county_name":"Cazengo"},
  {"id_county": 79,"id_province": 9,"county_name":"Golungo Alto"},
  {"id_county": 81,"id_province": 9,"county_name":"Lucala"},
  {"id_county": 82,"id_province": 9,"county_name":"Quiculungo"},
  {"id_county": 83,"id_province": 9,"county_name":"Samba Cajú"},
  {"id_county": 84,"id_province": 8,"county_name":"Amboim"},
  {"id_county": 85,"id_province": 8,"county_name":"Cassongue"},
  {"id_county": 86,"id_province": 8,"county_name":"Cela"},
  {"id_county": 87,"id_province": 8,"county_name":"Conda"},
  {"id_county": 88,"id_province": 8,"county_name":"Ebo"},
  {"id_county": 89,"id_province": 8,"county_name":"Libolo"},
  {"id_county": 90,"id_province": 8,"county_name":"Mussende"},
  {"id_county": 91,"id_province": 8,"county_name":"Porto Amboim"},
  {"id_county": 92,"id_province": 8,"county_name":"Quibala"},
  {"id_county": 93,"id_province": 8,"county_name":"Quilenda"},
  {"id_county": 94,"id_province": 8,"county_name":"Seles"},
  {"id_county": 95,"id_province": 8,"county_name":"Sumbe"},
  {"id_county": 96,"id_province": 1,"county_name":"Belas"},
  {"id_county": 97,"id_province": 1,"county_name":"Cacuaco"},
  {"id_county": 98,"id_province": 1,"county_name":"Cazenga"},
  {"id_county": 99,"id_province": 1,"county_name":"Ícolo e Bengo"},
  {"id_county": 100,"id_province": 1,"county_name":"Luanda"},
  {"id_county": 101,"id_province": 1,"county_name":"Quiçama"},
  {"id_county": 102,"id_province": 1,"county_name":"Viana"},
  {"id_county": 103,"id_province": 12,"county_name":"Dundo"},
  {"id_county": 104,"id_province": 12,"county_name":"Cambulo"},
  {"id_county": 105,"id_province": 12,"county_name":"Capenda-Camulemba"},
  {"id_county": 106,"id_province": 12,"county_name":"Caungula"},
  {"id_county": 107,"id_province": 12,"county_name":"Chitato"},
  {"id_county": 108,"id_province": 12,"county_name":"Cuango"},
  {"id_county": 109,"id_province": 12,"county_name":"Cuílo"},
  {"id_county": 110,"id_province": 12,"county_name":"Lubalo"},
  {"id_county": 111,"id_province": 12,"county_name":"Lucapa"},
  {"id_county": 112,"id_province": 12,"county_name":"Xá-Muteba"},
  {"id_county": 113,"id_province": 13,"county_name":"Cacolo"},
  {"id_county": 114,"id_province": 13,"county_name":"Dala"},
  {"id_county": 115,"id_province": 13,"county_name":"Muconda"},
  {"id_county": 116,"id_province": 13,"county_name":"Saurimo"},
  {"id_county": 117,"id_province": 14,"county_name":"Cacuso"},
  {"id_county": 118,"id_province": 14,"county_name":"Calandula"},
  {"id_county": 119,"id_province": 14,"county_name":"Cambundi Catembo"},
  {"id_county": 120,"id_province": 14,"county_name":"Cangandala"},
  {"id_county": 121,"id_province": 14,"county_name":"Caombo"},
  {"id_county": 122,"id_province": 14,"county_name":"Cuaba Nzogo"},
  {"id_county": 123,"id_province": 14,"county_name":"Cunda-Dia-Baze"},
  {"id_county": 124,"id_province": 14,"county_name":"Luquembo"},
  {"id_county": 125,"id_province": 14,"county_name":"Malanje"},
  {"id_county": 126,"id_province": 14,"county_name":"Marimba"},
  {"id_county": 127,"id_province": 14,"county_name":"Massango"},
  {"id_county": 128,"id_province": 14,"county_name":"Mucari"},
  {"id_county": 129,"id_province": 14,"county_name":"Quela"},
  {"id_county": 130,"id_province": 14,"county_name":"Quirima"},
  {"id_county": 131,"id_province": 15,"county_name":"Alto Zambeze"},
  {"id_county": 132,"id_province": 15,"county_name":"Bundas"},
  {"id_county": 133,"id_province": 15,"county_name":"Camanongue"},
  {"id_county": 134,"id_province": 15,"county_name":"Léua"},
  {"id_county": 135,"id_province": 15,"county_name":"Luau"},
  {"id_county": 136,"id_province": 15,"county_name":"Luacano"},
  {"id_county": 137,"id_province": 15,"county_name":"Luchazes"},
  {"id_county": 138,"id_province": 15,"county_name":"Luena"},
  {"id_county": 139,"id_province": 15,"county_name":"Lumeje"},
  {"id_county": 140,"id_province": 15,"county_name":"Moxico"},
  {"id_county": 141,"id_province": 16,"county_name":"Bibala"},
  {"id_county": 142,"id_province": 16,"county_name":"Camucuio"},
  {"id_county": 143,"id_province": 16,"county_name":"Moçâmedes"},
  {"id_county": 144,"id_province": 16,"county_name":"Tômbua"},
  {"id_county": 145,"id_province": 16,"county_name":"Virei"},
  {"id_county": 146,"id_province": 17,"county_name":"Alto Cauale"},
  {"id_county": 147,"id_province": 17,"county_name":"Ambuíla"},
  {"id_county": 148,"id_province": 17,"county_name":"Bembe"},
  {"id_county": 149,"id_province": 17,"county_name":"Buengas"},
  {"id_county": 150,"id_province": 17,"county_name":"Bungo"},
  {"id_county": 151,"id_province": 17,"county_name":"Damba"},
  {"id_county": 152,"id_province": 17,"county_name":"Macocola"},
  {"id_county": 153,"id_province": 17,"county_name":"Milunga"},
  {"id_county": 154,"id_province": 17,"county_name":"Mucaba"},
  {"id_county": 155,"id_province": 17,"county_name":"Negage"},
  {"id_county": 156,"id_province": 17,"county_name":"Puri"},
  {"id_county": 157,"id_province": 17,"county_name":"Quimbele"},
  {"id_county": 158,"id_province": 17,"county_name":"Quitexe"},
  {"id_county": 159,"id_province": 17,"county_name":"Sanza Pombo"},
  {"id_county": 160,"id_province": 17,"county_name":"Songo"},
  {"id_county": 161,"id_province": 17,"county_name":"Uíge"},
  {"id_county": 162,"id_province": 17,"county_name":"Zombo"},
  {"id_county": 163,"id_province": 18,"county_name":"Cuimba"},
  {"id_county": 164,"id_province": 18,"county_name":"M'Banza Kongo"},
  {"id_county": 165,"id_province": 18,"county_name":"Noqui"},
  {"id_county": 166,"id_province": 18,"county_name":"N'Zeto"},
  {"id_county": 167,"id_province": 18,"county_name":"Soyo"},
  {"id_county": 168,"id_province": 18,"county_name":"Tomboco"},
  {"id_county": 169,"id_province": 1,"county_name":"Kilamba Kiaxi"},
  {"id_county": 170,"id_province": 1,"county_name":"Talatona"},
  {"id_county": 171,"id_province": 9,"county_name":"Cazengo"},
  {"id_county": 172,"id_province": 9,"county_name":"Ngonguembo"},
  {"id_county": 173,"id_province": 14,"county_name":"Cahombo"},
  {"id_county": 174,"id_province": 14,"county_name":"Kunda Dya Baze"}
];
